infra/modules/providers/azure/provider/main.tf

```diff
@@ -1,5 +1,5 @@
 provider "azurerm" {
-  version = "~>1.32.0"
+  version = "~>1.34.0"
```

Comment:
Is there a changelog available for this? I'm wondering if we need to be on the lookout for any new breaks related to a provider rev.

Reply:
Good point. If the upstream is following SemVer, that change should t be breaking (or else the major number would have been incremented).

But, I don’t that they’re using SemVer. I can help check tomorrow, @KeithJRome

Reply:
https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/CHANGELOG.md#1340-september-18-2019

Reply:
I upgraded specifically because of a bug related to the provider code that was fixed -- https://github.com/terraform-providers/terraform-provider-azurerm/issues/4184

Reply:
Note that this bumps the Terraform dependency to 12.8 also (https://github.com/terraform-providers/terraform-provider-azurerm/pull/4341)

These changes rolled up into 1.34 look like they are good for us on balance, but I see several behavioral changes that we probably need to be aware of like how certain errors are being reported and some workarounds we have in place now may no longer be needed.

Reply:
Can you elaborate on that @KeithJRome ?

Reply:
@KeithJRome I read that thread as they are bumping the dependency to 0.12.8. Do you think that means anyone with the 1.34 provider needs to be running on 0.12.8? or is that only for the provider team? @nmiodice did you upgrade TF to 0.12.8? I didn't see that in the commits...

Reply:
I looked more closely into each of the changes since 1.32 and the one I was most interested in happens to be the same PR that fixes what you need. The others are very unlikely to cause us problems not that I've dug deeper into the actual code changes involved.

So I don't think this should cause us any problems. Carry On :)

Reply:
Yes @iphilpot, I think that means we need to rev our direct dependency on Terraform to 0.12.8, but I don't think this will cause us any problems. A few weird error messages might become less weird, and that's probably the extent of impact for us (aside from the bug that @nmiodice referred to being fixed).

Reply:
@KeithJRome @iphilpot I don't think that we should be driving the version of Terraform based off of the version of the library that the providers consume as its not currently possible to satisfy each provider.

Null Resource Provider: https://github.com/terraform-providers/terraform-provider-null/blob/b8568b9b73a19322eaf0c44982f3834ff34854ff/go.mod#L5

Azure AD Resource Provider: https://github.com/terraform-providers/terraform-provider-azuread/blob/0ed698531d3dd13fed4e51469fda2d517f98aa9d/go.mod#L10